wallet/walletrpc.c

```diff
@@ -393,6 +393,102 @@ static const struct json_command newaddr_command = {
 };
 AUTODATA(json_command, &newaddr_command);
 
+static void json_listaddrs(struct command *cmd,
+						   const char *buffer, const jsmntok_t *params)
+{
+	struct json_result *response = new_json_result(cmd);
+	struct ext_key ext;
+	struct sha256 h;
+	struct ripemd160 h160;
+	struct pubkey pubkey;
+	jsmntok_t *bip32tok;
+	u8 *redeemscript;
+	bool ok;
+	char *out_p2sh;
+	char *out_p2wpkh;
+	const char *hrp;
+	u64 bip32_max_index;
+	const tal_t *tmpctx = tal_tmpctx(NULL);
+
+	if (!json_get_params(cmd, buffer, params,
+						 "?bip32_max_index", &bip32tok, NULL)) {
+		tal_free(tmpctx);
+		return;
+	}
+
+	if (!bip32tok || !json_tok_u64(buffer, bip32tok, &bip32_max_index)) {
+		bip32_max_index = db_get_intvar(cmd->ld->wallet->db, "bip32_max_index", 0);
+	}
+	json_object_start(response, NULL);
+	json_array_start(response, "addresses");
+
+	for (s64 keyidx = 0; keyidx < bip32_max_index; keyidx++) {
+
+
+		if(keyidx == BIP32_INITIAL_HARDENED_CHILD){
+			command_fail(cmd, "Keys exhausted ");
+			tal_free(tmpctx);
+			return;
+		}
+
+		if (bip32_key_from_parent(cmd->ld->wallet->bip32_base, keyidx,
+								  BIP32_FLAG_KEY_PUBLIC, &ext) != WALLY_OK) {
```

Comment:
Please indent to align just after the opening `(` of the call.  Tabs are eight spaces.